adapters/arpalazio.js

```diff
@@ -100,89 +96,85 @@ const getParameters = function (averagingPeriod) {
   }
 };
 
-const getData = function (cityName, url, averagingPeriod, source) {
-  return function (done) {
-    const match = url.match(/[\w]{2}_([\w.]{2,})_([\d]{4})(?:_gg)?.txt/);
-    const parameter = match[1].toLowerCase().replace('.', '');
-    const year = match[2];
-    const unit = getUnit(parameter);
-
-    request(url, (err, res, body) => {
-      if (err || res.statusCode !== 200) {
-        return done(null, []);
-      }
-      // remove whitespace for clean parsing
-      body = body.replace(/[ ]+/g, ' ');
-      body = body.replace(/^[ ]/g, '');
-      body = body.replace(/\n[ ]+/g, '\n');
-
-      const parsed = parse(body, {delimiter: ' '});
-      const headers = parsed.slice(0, 1)[0];
-      const records = parsed.slice(1);
-
-      // zip them to check -999 values
-      let zipped = zip(...records);
-      let invalidStationIDs = [];
-      zipped.forEach(function (values, i, _) {
-        if (values.every((v) => Number(v) === -999)) {
-          invalidStationIDs.push(headers[i]);
-        }
-      });
-
-      let measurements = [];
-
-      const fewDaysAgo = moment.tz(timezone).subtract(4, 'days');
-      const recentRecords = records.filter((row) => Number(row[headers.indexOf('jd')]) >= Number(fewDaysAgo.format('DDD')));
-      recentRecords.forEach(function (row) {
-        let date;
-        if (averagingPeriod.value === 1) {
-          date = moment.tz(`${year} ${row[0]} ${row[1]}`, 'YYYY DDD HH', timezone);
-        } else {
-          date = moment.tz(`${year} ${row[1]}`, 'YYYY DDD', timezone);
-        }
-
-        const fullDate = {
-          utc: date.toDate(),
-          local: date.format()
+export const getStream = function (cityName, url, averagingPeriod, source, orgUrl) {
+  const { metadata } = source;
+  const match = url.match(/[\w]{2}_([\w.]{2,})_([\d]{4})(?:_gg)?.txt/);
+  const parameter = match[1].toLowerCase().replace('.', '');
+  const year = match[2];
+  const unit = getUnit(parameter);
+  const dayPosition = averagingPeriod.value === 1 ? 0 : 1;
```

Comment:
```suggestion
    const match = url.match(/[\w]{2}_([\w.]{2,})_([\d]{4})(_gg)?.txt/);
    const parameter = match[1].toLowerCase().replace('.', '');	  const parameter = match[1].toLowerCase().replace('.', '');
    const year = match[2];	  const year = match[2];
    const unit = getUnit(parameter);	  const unit = getUnit(parameter);
    const dayPosition = match[3] === '_gg' ? 0 : 1;
```

Reply:
@jflasher The above suggestion is for the whole block from L101 to L105, not the single one.

Reply:
@MichalCz if I'm reading this correctly, you're changing how `dayPosition` is calculated? I think is probably fine if it looks good in the resultant data? All of this scraping is a bit hacky anyways.